builder/docker/communicator.go

```diff
@@ -18,12 +18,13 @@ import (
 )
 
 type Communicator struct {
-	ContainerID  string
-	HostDir      string
-	ContainerDir string
-	Version      *version.Version
-	Config       *Config
-	lock         sync.Mutex
+	ContainerID   string
+	HostDir       string
+	ContainerDir  string
+	Version       *version.Version
+	Config        *Config
+	containerUser *string
```

Comment:
Let's use a non-pointer string here, unless there's a specific reason to I'm not seeing. Can check for an empty string `... == ""` to see if it's set or not.

Reply:
The pointer is used to run `docker inspect` only once. `docker inspect` returns empty string if `USER` has not been set so I needed a different sentinel to mark that `docker inspect` has already run.

However, I can think of a better approach here, a more correct one. To use a string - not a pointer - and when `docker inspect` returns an empty string set `containerUser` to `root`(if that is a safe assumption?). This is better because (1) no pointer and (2) current solution does not change owner when `docker inspect` returns an empty string. So when container is run as root, uploaded files will be left owned by the user set by `docker cp`, which might be unexpected.

Reply:
okay, I see what you were doing. Apologies for not mentioning it earlier, but putting the set-up code in `builder/docker/step_connect_docker.go` would be best. That way, you can make sure you only `fixDestinationOwner` once. That logic probably doesn't even need to be in the communicator... in `step_connect_docker.go`, you can check the flag if we're going to auto-fix the user, and if so, run `exec.Command` and get the user directly. 

Does the above make sense? I can help more if you want